### PR TITLE
use split_once for cleaner code

### PR DESCRIPTION
**Pull request by Eh2406**
### What does this PR try to resolve?

Search the code base for `.splitn(2` and replace with `.split_once` where it was clearer. I don't think any of them matter in practice.

### How should we test and review this PR?

This was an internal re-factor, and the tests still pass.
The two methods have subtly different semantics, so please review carefully.

### CODE REVIEW

File: src/cargo/core/package_id_spec.rs [hunk 1]
```diff
@@ -123,24 +123,20 @@ impl PackageIdSpec {
...
-                        }
-                        None => {
-                            if name_or_version.chars().next().unwrap().is_alphabetic() {
-                                (InternedString::new(name_or_version), None)
-                            } else {
-                                let version = name_or_version.to_semver()?;
-                                (InternedString::new(path_name), Some(version))
-                            }
+                Some(fragment) => match fragment.split_once([':', '@']) {
+                    Some((name_or_version, part)) => {
```
Comment (epage) [inline src/cargo/core/package_id_spec.rs:127]: I believe `name_or_version` is only `name` here

**Review by epage — APPROVED** [review]

**Comment by epage** [comment]
Feel free to `r=` me whether you decide to resolve the nit or not

File: crates/mdman/src/main.rs [hunk 2]
```diff
@@ -98,15 +98,16 @@ fn process_args() -> Result<Options, Error> {
                 let man = args
                     .next()
                     .ok_or_else(|| format_err!("--man requires a value"))?;
-                let parts: Vec<_> = man.splitn(2, '=').collect();
-                let key_parts: Vec<_> = parts[0].splitn(2, ':').collect();
-                if parts.len() != 2 || key_parts.len() != 2 {
-                    bail!("--man expected value with form name:1=link");
-                }
```
Comment (Eh2406) [inline crates/mdman/src/main.rs:105]: The old code errored on having too many `=` or `:` which the new code no longer does. Do we care?

File: src/cargo/core/package_id_spec.rs (hunk 1 above)
Comment (Eh2406) [reply src/cargo/core/package_id_spec.rs]: Done.

File: crates/mdman/src/main.rs (hunk 2 above)
Comment (weihanglo) [reply crates/mdman/src/main.rs:105]: I don't fully understand the format but we can keep the behavior the same for this refactor

**Review by weihanglo — APPROVED** [review]

**Pull request merged**